plugins/explore-backend/src/tools/types.ts

```diff
@@ -31,3 +32,17 @@ export interface ExploreToolProvider {
    */
   getTools(request: GetExploreToolsRequest): Promise<GetExploreToolsResponse>;
 }
+
+/** @public */
+export interface ExploreProviderRegistrationOptions {
+  providerId: string;
+  factory: StaticExploreToolProvider;
+}
+
+/** @public */
+export interface ExporeProvidersExtensionPoint {
+  registerProvider(options: ExploreProviderRegistrationOptions): void;
+}
+
+/** @public */
+export type ProviderFactories = { [s: string]: StaticExploreToolProvider };
```

Comment:
Why is this type pointing to the literal `StaticExploreToolProvider`? This type is an object with any number of keys, all pointing to exactly the static tool provider. Also not sure why it's an object in the first place, can there be many?